ctx.go

```diff
@@ -753,9 +763,11 @@ func (ctx *Ctx) Range(size int) (rangeData Range, err error) {
 		})
 	}
 	if len(rangeData.Ranges) < 1 {
-		return rangeData, fmt.Errorf("range: unsatisfiable range")
+		err = ErrRangeUnsatisfiable
```

Comment:
Why not return `rangeData, ErrRangeUnsatisfiable` here

Reply:
> Why not `return rangeData, ErrRangeUnsatisfiable` here 

@songjiayang I think this will reassign rangeData again and I like unifying the code style :smile:

Reply:
If you return the error directly at above code it looks unifying yet 😄 , for example  `return nil,  ErrRangeMalformed`

`No Need Change`  it's personal behavior

Reply:
When using `named return vars`, I prefer the `empty return` way.